### ISSUE TITLE
Framing issues rsyslog (Bluecoat log forwarding to syslog collector)

### ISSUE THREAD

**Issue by yanish48** [rsyslog/rsyslog#5128]
I am bringing data from a log source and by reviewing I found that it is broken into multiple lines. 
### Expected behavior
Log should be parsed properly and forwarded to SIEM.

### Actual behavior
Giving errors like
imtcp imtcp: Framing Error in received TCP message from peer: (hostname) 192.168.14.71, (ip) 192.168.14.71: delimiter is not SP but has ASCII value 47. [v8.2001.0]
The ASCII values found in the error message are 45,46,47,56,58,98,99
I have tried adding the parameters found on other similar errors:
InputTCPSupportOctetCountedFraming Off which gives error that  command 'InputTCPServerSupportOctetCountedFraming' is currently not permitted - did you already set it via a RainerScript command (v6+ config)? [v8.2001.0 try https://www.rsyslog.com/e/2222 ]
and invalid or yet-unknown config file command 'SupportOctetCountedFraming' - 

### Environment
- rsyslog version:  8.2001.0 
- platform:  Ubuntu 20.04
- for configuration questions/issues, include rsyslog.conf and included config files

# Default logging rules can be found in /etc/rsyslog.d/50-default.conf

#################
#################

module(load="imuxsock") # provides support for local system logging
#module(load="immark")  # provides --MARK-- message capability

# provides UDP syslog reception
module(load="imudp")
input(type="imudp" port="514")

# provides TCP syslog reception
module(load="imtcp")
input(type="imtcp" port="514")

# provides kernel logging support and enable non-kernel klog messages
module(load="imklog" permitnonkernelfacility="on")

###########################
###########################

#
# Use traditional timestamp format.
# To enable high precision timestamps, comment out the following line.
#
$ActionFileDefaultTemplate RSYSLOG_TraditionalFileFormat

# Filter duplicated messages
$RepeatedMsgReduction on

$SupportOctetCountedFraming Off
#
# Set the default permissions for all log files.
#
$FileOwner syslog
$FileGroup adm
$FileCreateMode 0640
$DirCreateMode 0755
$Umask 0022
$PrivDropToUser syslog
$PrivDropToGroup syslog

#
# Where to place spool and state files
#
$WorkDirectory /var/spool/rsyslog

**Comment by davidelang** [comment]
please log a message with the template RSYSLOG_DebugFormat or capture it via 
tcpdump.

it sounds as if you are being sent a message that's not formatted properly per 
the syslog RFCs

David Lang

**Comment by yanish48** [comment]
Yes David, I am wondering the same. Any recommendations/workarounds for the same?
Please find below the set of sample logs:
Apr 27 00:00:24 192.168.14.71 ure" - 192.168.13.71 2 #015

Apr 27 00:00:24 04-27 00: 02:56 110 192.168.244.39 200 TCP_NC_MISS 542 88 GET http www.msftconnecttest.com 80 /connecttest.txt - - - www.msftconnecttest.com 13.107.4.52 "None" - text/plain;%20charset=utf-8 - - OBSERVED "Web Infrastructure" - 192.168.13.71 2 #015#0122023-04-27 00:02:56 25243 192.168.32.52 200 TCP_NC_MISS 271 488 POST http 149.154.167.91 80 /api - - - 149.154.167.91 149.154.167.91 "None" - application/octet-stream - "Mozilla/5.0" OBSERVED "Chat (IM)/SMS" - 192.168.13.71 5 #015#0122023-04-27 00:02:56 194 192.168.173.68 200 TCP_NC_MISS 129 142 POST http 77.68.91.42 80 / - - - 77.68.9
11:25:10 
	1.42 77.68.91.42 "None" - - - - OBSERVED "none" - 192.168.13.71 4 #015#0122023-04-27 00:02:56 151 192.168.32.52 200 TCP_NC_MISS 271 488 POST http 149.154.167.91 80 /api - - - 149.154.167.91 149.154.167
	.91 "None" - application/octet-stream - "Mozilla/5.0" OBSERVED "Chat (IM)/SMS" - 192.168.13.71 5 #015#0122023-04-27 00:02:56 1 192.168.40.71 403 TCP_DENIED 13325 287 GET http ctldl.windowsupdate.com 80
	 /msdownload/update/v3/static/trustedr/en/disallowedcertstl.cab ?5c64b2f4731a8e35 - - - - "None" - - - "Microsoft-CryptoAPI/10.0" DENIED "Microsoft Intune;Area windowsupdate Blocked URL;Web Infrastruct
	ure" - 192.168.13.71 1 #015#0122023-04-27 00:02:56 1 192.168.11.3 400 TCP_NC_MISS 1165 9 GET http 192.168.11.71 80 / - - - 192.168.11.71 192.168.11.71 "Unavailable" - - - - DENIED "none" - 192.168.11.7
	1 none #015#0122023-04-27 00:02:57 25208 192.168.32.63 200 TCP_NC_MISS 271 459 POST http 149.154.175.60 80 /api - - - 149.154.175.60 149.154.175.60 "None" - application/octet-stream - "Mozilla/5.0" OBS
11:25:26 From  j&m  to  Everyone:
	1 none #015#0122023-04-27 00:02:57 25208 192.168.32.63 200 TCP_NC_MISS 271 459 POST http 149.154.175.60 80 /api - - - 149.154.175.60 149.154.175.60 "None" - application/octet-stream - "Mozilla/5.0" OBS
	ERVED "Chat (IM)/SMS" - 192.168.13.71 2 #015#0122023-04-27 00:02:57 111 192.168.244.39 200 TCP_NC_MISS 542 88 GET http www.msftconnecttest.com 80 /connecttest.txt - - - www.msftconnecttest.com 13.107.4
	.52 "None" - text/plain;%20charset=utf-8 - - OBSERVED "Web Infrastructure" - 192.168.13.71 2 #015#0122023-04-27 00:02:57 1 192.168.120.139 403 TCP_DENIED 13331 202 GET http ctldl.windowsupdate.com 80 /
	msdownload/update/v3/static/trustedr/en/disallowedcertstl.cab ?cbfe37f1f18adcf

Apr 27 00:00:24 192.168.14.71 - - "None" - - - "Microsoft-CryptoAPI/10.0" DENIED "Microsoft Intune;Area windowsupdate Blocked URL;Web Infrastructure" - 192.168.13.71 1 #015 11:30:36

**Comment by davidelang** [comment]
what we need to see are the raw lots, which is why I asked you to log the 
messages with the template RSYSLOG_DebugFormat

it looks like the log entries are being sent with a carriage return/line feed 
rather than just the linefeed, but something else is happening to cause rsyslog 
to not treat the linefeed as the end of the log (leaving you with a stray 
carriage return in the log message).

please post your full config.

David Lang
